tools/cli/README.md

```diff
@@ -0,0 +1,71 @@
+## What
+Cadence CLI provide an command-line tool for users to perform various tasks on Cadence.  
+Users can perform operations like register, update and describe on domain;  
+also start workflow, show workflow history, signal workflow ... and many other tasks.    
+
+## How
+- Run `make bins`
+- You should see an executable `cadence-cli`
+
+## Quick Start
+Run `./cadence-cli` to view help message. There are some top level commands and global options.   
+Run `./cadence-cli domain` to view help message about operations on domain  
+Run `./cadence-cli workflow` to view help message about operations on workflow  
+
+*Note:* make sure you have cadence server running before using CLI 
+
+### Domain operation examples 
+- Register a new domain named "samples-domain":  
+```
+./cadence-cli --domain samples-domain domain register 
+# OR using short alias  
+./cadence-cli --do samples-domain domain re
+```   
+- View "samples-domain" details:   
+```
+./cadence-cli --domain samples-domain domain describe  
+```
+
+### Workflow operation examples
+
+- Start workflow: (for example [this workflow](https://github.com/samarabbas/cadence-samples/blob/master/cmd/samples/recipes/helloworld/helloworld_workflow.go))
+```
+./cadence-cli --domain samples-domain workflow start --tl helloWorldGroup --wt main.Workflow --et 60 --dt 10 -i '"vancexu"'
+
+# see more options for workflow start
+./cadence-cli --domain samples-domain workflow start -h
+
+# for workflow with multiple input, seperate each json with space/newline like
+./cadence-cli --domain samples-domain workflow start --tl helloWorldGroup --wt main.WorkflowWith3Args --et 60 --dt 10 -i '"your_input_string" 123 {"Name":"my-string", "Age":12345}'
+```
+
+- Run workflow and see the progress 
```

Comment:
I think this one should come before start as most developers should use it during development. I think you also want to expand on what "run" means in this context and what benefits over start it has.

Reply:
changed